docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,343 @@
+
+```bash
+level=info ts=2018-12-05T07:00:08.164203448Z caller=compact.go:398 component=tsdb msg="write block" mint=1543982400000 maxt=1543989600000 ulid=01CXYJN992AVWSVVJFP6X4E8MR
+level=info ts=2018-12-05T07:00:09.142253971Z caller=head.go:446 component=tsdb msg="head GC completed" duration=183.072583ms
+
+level=info ts=2018-12-05T09:00:08.289897499Z caller=compact.go:398 component=tsdb msg="write block" mint=1543989600000 maxt=1543996800000 ulid=01CXYSH0M8RSWNJG3YM1S9YGY5
+level=info ts=2018-12-05T09:00:09.242667991Z caller=head.go:446 component=tsdb msg="head GC completed" duration=170.063092ms
+```
+
+You can see how the delta between those two compactions is two hours.
+
+In order to do a safe migration and avoid time range overlaps in later compactions, it is necessary to create a snapshot in Prometheus skipping the WAL file, as it contains references to future time ranges.
+
+It is important to know that if you create a Prometheus snapshot skipping the WAL file the **last hour of data** will not be present in the snapshot. For example, if the Prometheus logs show the following operations:
+
+```bash
+level=info ts=2018-12-05T11:00:08.289897499Z caller=compact.go:398 component=tsdb msg="write block" mint=1543989600000 maxt=1543996800000 ulid=01CXYSH0M8RSWNJG3YM1S9YGY5
+level=info ts=2018-12-05T13:00:09.242667991Z caller=head.go:446 component=tsdb msg="head GC completed" duration=170.063092ms
+```
+
+A good time to create a snapshot of the TSDB  to perform the migration would be at 13:01 UTC, so you create the snapshot just after Prometheus wrote the latest information stored in the WAL file to disk.
+If you do this, you will find the following:
+
+[Insert image of  Prometheus query with 1 hour of missing data due to snapshot]
+
+You would have lost 1 hour of Prometheus metrics.
+
+In order to ensure a minimal data loss, the new Prometheus deployment should be deployed 55 mins (ensuring with this less than 5 mins of data loss) before the next TSDB compaction happens, so the new Prometheus deployment can start scraping data before the snapshot is restored. Continuing with the last example:
+
+If the next TSDB compaction happens at 13:00 UTC you should deploy BKPR Prometheus in the cluster at 12:05 UTC, so Prometheus can start scraping data and fill the gap that the snapshot will create. Then, at 13:00 UTC you can create the Prometheus snapshot and restore it in the new deployment, creating the following:
+
+![prometheus_successful_migration](../../images/prometheus-migration/tsdb_restored.png)
+
+## Step 4: Create a TSDB snapshot and restore it in the new deployment
+
+Now that the process has been explained in detail, these are the commands that will have to be executed in order to migrate the TSDB from the old Prometheus deployment to the BKPR one.
```

Comment:
commands that will have to be executed -> commands to be executed

Reply:
done